wicker/core/datasets.py

```diff
@@ -303,3 +452,85 @@ def dataset_size(self) -> int:
             int: total dataset size in bytes
         """
         return self._get_dataset_partition_size()
+
+
+class DatasetFactory:
```

Comment:
Thanks a lot for adding this, I think a function is enough, we do not sub class the factory since the dependency injection is not so complex. 
I am fine with a factory class if you think we will sub class eventually